pkg/cmd/server/kubernetes/master/controller/config.go

```diff
@@ -36,48 +23,14 @@ func (c KubeControllerConfig) GetControllerInitializers() (map[string]kubecontro
 	// in openshift-infra, and pass it a scale client that knows how to scale DCs
 	ret["horizontalpodautoscaling"] = c.HorizontalPodAutoscalerControllerConfig.RunController
 
-	// FIXME: Move this under openshift controller intialization once we figure out
-	// deployment (options).
-	ret["openshift.io/scheduler"] = c.SchedulerControllerConfig.RunController
-
 	return ret, nil
 }
 
 // BuildKubeControllerConfig builds the struct to create the controller initializers.  Eventually we want this to be fully
 // stock kube with no modification.
 func BuildKubeControllerConfig(options configapi.MasterConfig) (*KubeControllerConfig, error) {
-	var err error
 	ret := &KubeControllerConfig{}
 
-	kubeExternal, _, err := configapi.GetExternalKubeClient(options.MasterClients.OpenShiftLoopbackKubeConfig, options.MasterClients.OpenShiftLoopbackClientConnectionOverrides)
```

Comment:
I dont see we are making sure now that connection overrides still apply, since now the client will be created in scheduler itself, are we ok with this?

```
// applyClientConnectionOverrides updates a kubeConfig with the overrides from the config.
func applyClientConnectionOverrides(overrides *ClientConnectionOverrides, kubeConfig *restclient.Config) {
        if overrides == nil {
                return
        }
        kubeConfig.QPS = overrides.QPS
        kubeConfig.Burst = int(overrides.Burst)
        kubeConfig.ContentConfig.AcceptContentTypes = overrides.AcceptContentTypes
        kubeConfig.ContentConfig.ContentType = overrides.ContentType
}
```

Reply:
> I dont see we are making sure now that connection overrides still apply, since now the client will be created in scheduler itself, are we ok with this?

Yes.  The path forward will be to configure the differences we need in the scheduler CLI.  If the scheduler doesn't support it feature we need, we'll be pursuing them upstream and perhaps picking them back.

Reply:
Thats ok I agree with this. I am concerned if someone supplied QPS/Burst/ContentType in openshift, and doesn't realize that these values wont be passed to scheduler any more? wouldn't it create regressions?

Reply:
> Thats ok I agree with this. I am concerned if someone supplied QPS/Burst/ContentType in openshift, and doesn't realize that these values wont be passed to scheduler any more? wouldn't it create regressions?

Are the defaults bad?  If the defaults upstream are good, I'm probably willing to crack some eggs.  If they're bad, we do some shenanigans to override.  The overrides in openshift are pretty coarse grained and targeted at loopback (which we'll probably be disabling) and controllers (which we have to keep).

Just remember that everything we plumb through special is another thing you'll be describing to ansible to special-case during the (hopefully) 3.8 migration.

Reply:
I am not really sure upstream defaults are good or bad, lets say they are good. But I agree with your ansible thing.

But It is getting a bit confusing now as there are 4 ways to set QPS/Burst etc
1. Via kubeconfig
2. Via overrides
3. Via scheduler's defaults (https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1/defaults.go#L139)
4. Via scheduler server's flags

Lets say overrides are not in picture any more, I think I still see issue in createClient function:  https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/plugin/cmd/kube-scheduler/app/configurator.go#L57

Even if a user does not set any scheduler flags for QPS/burst etc, QPS/Burst kubeconfig values are being overwritten by defaults, as there is no way to differentiate if these were set by user or are defaults. 

I think we will have to decide if we are ok with kubeconfig supplied values being overwritten by defaults.  I think this issue is something I will have to look into kube upstream too.

Though, one thing I am wondering, I am not sure, in practice what is preferred way of configuration for these QPS/Burst, via kubeconfig, or via scheduler's flags etc.